src/Microsoft.NET.Sdk.Functions.Generator/FunctionJsonConverter.cs

```diff
@@ -173,36 +175,86 @@ private bool TryGenerateFunctionJsons()
-                            exportedTypes = exportedTypes.Concat(loadedModule.Types);
-                        }
-                        catch (Exception ex)
+                        var tryPath = Path.Combine(assemblyRoot, $"{referencedAssembly.Name}.dll");
+                        if (File.Exists(tryPath))
                         {
-                            _logger.LogWarning($"Could not evaluate '{referencedAssembly.Name}' for function types. Exception message: {ex.Message}");
+                            try
+                            {
+                                var loadedModule = ModuleDefinition.ReadModule(tryPath, readerParams);
+                                exportedTypes = exportedTypes.Concat(loadedModule.Types);
+                            }
+                            catch (Exception ex)
+                            {
+                                _logger.LogWarning($"Could not evaluate '{referencedAssembly.Name}' for function types. Exception message: {ex.Message}");
+                            }
                         }
                     }
                 }
+
+                var functions = GenerateFunctions(exportedTypes).ToList();
+                foreach (var function in functions.Where(f => f.HasValue && !f.Value.outputFile.Exists).Select(f => f.Value))
+                {
+                    function.schema.Serialize(function.outputFile.FullName);
+                }
+                ret = functions.All(f => f.HasValue);
             }
 
-            var functions = GenerateFunctions(exportedTypes).ToList();
-            foreach (var function in functions.Where(f => f.HasValue && !f.Value.outputFile.Exists).Select(f => f.Value))
+            loadContext.Unload();
+            return ret;
+
+            Assembly ResolveAssembly(AssemblyLoadContext context, AssemblyName assemblyName)
             {
-                function.schema.Serialize(function.outputFile.FullName);
+                Assembly result = null;
+
+                // First, check the assembly root dir. Assemblies copied with the app always wins.
+                var path = Path.Combine(assemblyRoot, assemblyName.Name + ".dll");
+                if (File.Exists(path))
+                {
+                    result = context.LoadFromAssemblyPath(path);
+
+                    if (result != null)
+                    {
+                        return result;
+                    }
+                }
+
+                // Then, check if the assembly is already loaded into the default context.
+                // This is typically the case for framework assemblies like System.Private.Corelib.dll.
+                if (AssemblyLoadContext.Default.Assemblies.FirstOrDefault(a => a.GetName().Equals(assemblyName)) is Assembly existing)
+                {
+                    path = existing.Location;
+                    result = context.LoadFromAssemblyPath(path);
```

Comment:
Is loading this again in the context an intentional decision? The typical case here will be for Fx assemblies, which we could just use from the default context (i.e., avoid the load and return the assembly match). Is there a strong reason for this approach? Reusing would also make the above point moot.

Reply:
My lack of knowledge of assembly load contexts. I was unsure if it was safe to do so, and decided to go with more loading rather than potential crashes.

Reply:
Back in front of a computer, so I can write a bit better than by phone 😛. Basically, I was worried about whether or not a resolve request by load context A that resolves to an assembly loaded by load context B could pose issues. If it can't, then the existing `Assembly` should just be returned.

Reply:
Particularly for those cases, this should be safe, so we should be able to just return the assembly we got from the default context (this would essentially unify on those, which is not super important in this particular context, but would be problematic if we were actually interacting with that code)

Reply:
Changed to return existing.